AppBar documentation (#1749)
telerik/blazor-docs: components/appbar/overview.md

---
title: Overview
page_title: AppBar Overview

> description: Overview of the AppBar component for Blazor.

Reviewer: ```suggestion
description: Discover the AppBar component for Blazor. Learn how to add the component to your app and explore its features like sections, dividers, positioning, and various styling options.
```

## Creating Blazor AppBar

1. Add the `<TelerikAppBar>` tag to a Razor file.
1. Use the `<AppBarSection>` child tag to add content to the AppBar component.
1. (optional) Use [spacers or separators](#content-dividers) to add visual distinction between the sections in the AppBar.

>caption Basic configuration of the Telerik AppBar

````CSHTML
<TelerikAppBar>
    <AppBarSection>
        <span>Company Logo</span>
    </AppBarSection>

    <AppBarSpacer></AppBarSpacer>

    <AppBarSection>
        <span>Our Products</span>
    </AppBarSection>

    <AppBarSeparator></AppBarSeparator>

    <AppBarSection>
        <span>Our Mission</span>
    </AppBarSection>

    <AppBarSeparator></AppBarSeparator>

    <AppBarSection>
        <span>Contact Us</span>
    </AppBarSection>

    <AppBarSpacer></AppBarSpacer>

    <AppBarSection>
        <TelerikSvgIcon Icon="@SvgIcon.User"></TelerikSvgIcon>
    </AppBarSection>

    <AppBarSeparator></AppBarSeparator>

    <AppBarSection>
        <TelerikSvgIcon Icon="@SvgIcon.Logout"></TelerikSvgIcon>
    </AppBarSection>
</TelerikAppBar>
````

## AppBar Sections

> Use the AppBar Sections to render arbitrary HTML content to match the UI and UX needs of your application. [Read more about the Blazor AppBar sections]({%slug appbar-sections%}).

Reviewer: Read more about... - use "..." as a part of the anchor link instead of "." This comment is valid for all sections with features.
```suggestion
Use the AppBar Sections to render arbitrary HTML content to match the UI and UX needs of your application. [Read more about the Blazor AppBar sections...]({%slug appbar-sections%})
```

| --- | --- | --- |
| `Class` | `string` | The CSS class to be rendered on the main wrapping element of the AppBar component, which is `<div class="k-appbar">`. Use for [styling customizations]({%slug themes-override%}). |
| `Height` | `string` | The height of the AppBar. |
| `ThemeColor` | `Telerik.Blazor.ThemeConstants.AppBar.ThemeColor` | Adjust the color of the AppBar |
| `Width` | `string` | The width of the AppBar. |

You can find more information for customizing the AppBar appearance in the [Appearance article]({%slug appbar-appearance%}).

## AppBar Reference and Methods

To execute AppBar methods, obtain reference to the component instance with `@ref`.

| Method  | Description |
|---------|-------------|
| `Refresh` | Use the method to programmatically re-render the AppBar. |

<div class="skip-repl"></div>

````CSHTML
<TelerikButton OnClick="@RefreshAppBar">Refresh AppBar</TelerikButton>

<TelerikAppBar @ref="AppBarRef" />

@code {
	private TelerikAppBar AppBarRef { get; set; }

	private void RefreshAppBar()
	{
		AppBarRef.Refresh();
	}
}
````

## Next Steps

* [Explore the AppBar Sections]({%slug appbar-sections%})
* [Use the AppBar Sections]({%slug appbar-separators%})

> * [Customize the AppBar position]({%slug appbar-position%})

Reviewer: ```suggestion
* [Customize the AppBar Position]({%slug appbar-position%})
```

---
title: Overview
page_title: AppBar Overview
description: Overview of the AppBar component for Blazor.
slug: appbar-overview
tags: telerik,blazor,appbar,navbar
published: True
position: 0
---

# Blazor AppBar Overview

> The <a href = "https://www.telerik.com/blazor-ui/appbar" target="_blank">Blazor AppBar component</a> helps you build navigation bars for your application seemingly. This article explains the available features.

Reviewer 2: ```suggestion
The <a href = "https://www.telerik.com/blazor-ui/appbar" target="_blank">Blazor AppBar component</a> helps you build navigation bars for your application seamlessly. This article describes the available features.
```

## Creating Blazor AppBar

1. Add the `<TelerikAppBar>` tag to a Razor file.
1. Use the `<AppBarSection>` child tag to add content to the AppBar component.

> (optional) Use [content dividers](#content-dividers) to add visual distinction between the sections in the AppBar.

Reviewer 2: ```suggestion
1. (optional) Use [spacers or separators](#content-dividers) to add visual distinction between the sections in the AppBar.
```

> 1. (optional) Set the [position](#positioning) of the AppBar.

Reviewer 2: Each step must be reflected in the code example afterwards. Either remove the step, or add the setting in the code.

Author: We have no good way to showcase the Postion/PositionMode in code snippets without iframe. So I will remove the step :)

>caption Basic configuration of the Telerik AppBar

````CSHTML
<TelerikAppBar>
    <AppBarSection>

>         <span>Our Logo</span>

Reviewer 2: ```suggestion
        <span>Company Logo</span>
```

    </AppBarSection>

>     <AppBarSpacer Size="25%"></AppBarSpacer>

Reviewer 2: The different ways to set `Size` make the empty spaces look uneven and the layout not so good. Better leave these details for the dedicated Separators article, and make the first impression better.

Also, consider `<AppBarSeparator />` between the three text items in the middle, which is more real-world too.

    <AppBarSection>
        <span>Our Products</span>
    </AppBarSection>

    <AppBarSpacer Size="50px"></AppBarSpacer>

    <AppBarSection>
        <span>Our Mission</span>
    </AppBarSection>

    <AppBarSpacer Size="5vw"></AppBarSpacer>

    <AppBarSection>
        <span>Contact Us</span>
    </AppBarSection>

    <AppBarSpacer></AppBarSpacer>

    <AppBarSection>
        <TelerikSvgIcon Icon="@SvgIcon.User"></TelerikSvgIcon>
    </AppBarSection>

    <AppBarSeparator></AppBarSeparator>

    <AppBarSection>
        <TelerikSvgIcon Icon="@SvgIcon.Logout"></TelerikSvgIcon>
    </AppBarSection>
</TelerikAppBar>
````

## AppBar Sections

> Use the AppBar Sections to render arbitrary HTML content to match the UI and UX needs of your application. [Read more about the Blazor AppBar sections]({%slug appbar-section%}).

Reviewer 2: ```suggestion
Use the AppBar Sections to render arbitrary HTML content to match the UI and UX needs of your application. [Read more about the Blazor AppBar sections]({%slug appbar-sections%}).
```

## Content Dividers

The AppBar features separators and spacers that can visually divide the component items. [Read more about the Blazor AppBar separators and spacers.]({%slug appbar-separators%}).

## Positioning

You can control the position of the AppBar and how the component behaves according to the flow of the page. [Read more about the Blazor AppBar positioning.]({%slug appbar-position%}).

## AppBar Parameters

> The Blazor AppBar provides parameters to configure the component:

Reviewer 2: LInk the API here too.

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Description |
| ----------- | ----------- | ----------- |
| `Class` | `string` | The CSS class to be rendered on the main wrapping element of the AppBar component, which is `<div class="k-appbar">`. Use for [styling customizations]({%slug themes-override%}). |
| `Height` | `string` | The height of the AppBar. |
| `Position` | `AppBarPosition` <br /> (`None`) | The position of the AppBar on the page. [Read more about AppBar positioning.]({%slug appbar-position%}) |
| `PositionMode` | `AppBarPosition` <br /> (`Static`) | Sets how the AppBar is positioned according to the flow of the document. [Read more about AppBar positioning.]({%slug appbar-position%}) |
| `Width` | `string` | The width of the AppBar |

> ### Styling and Appearance

Reviewer 2: If you wish to have a separate styling section, move `Class`, `Height` and `Width` there. Or do the opposite - move the `ThemeColor` to the main parameter table.

The following parameters enable you to customize the appearance of the Blazor AppBar:

| Parameter | Type | Description |
| --- | --- | --- |
| `ThemeColor` | `Telerik.Blazor.ThemeConstants.AppBar.ThemeColor` | Adjust the color of the AppBar |

You can find more information for customizing the AppBar appearance in the [Appearance article]({%slug appbar-appearance%}).

## AppBar Reference and Methods

> To execute AppBar methods, obtain reference to the component instance via `@ref`.

Reviewer 2: ```suggestion
To execute AppBar methods, obtain reference to the component instance with `@ref`.
```

| Method  | Description |
|---------|-------------|

> | `Refresh` | Use the method to programmatically re-render the AppBar. |

Reviewer 2: When will this be necessary?

<div class="skip-repl"></div>

````CSHTML
<TelerikButton OnClick="@RefreshAppBar">Refresh AppBar</TelerikButton>

<TelerikAppBar @ref="AppBarRef" />

@code {
	private TelerikAppBar AppBarRef { get; set; }

	private void RefreshAppBar()
	{
		AppBarRef.Refresh();
	}
}
````

## Next Steps

* [Explore the AppBar Sections]({%slug appbar-sections%})

> * [Use the AppBar Content]({%slug appbar-separators%})

Reviewer 2: The link text ("content") doesn't relate to the target page.
